backend/app/services/advanced_analytics_service.py

```
@@ -196,7 +209,9 @@
             )
 
         except Exception as e:
-            logger.error(f"Business intelligence report generation failed: {str(e)}")
+            logger.error(
+                f"Business intelligence report generation failed: {str(e)}"
+            )
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)